osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -36,30 +45,84 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GridMeasurementPublishedEventConverter.class);
 
-    private static final int SIMPLE_END_INDEX = 10;
+    private ObjectMapper objectMapper = new ObjectMapper();
 
     @Override
     public GridMeasurementPublishedEvent convert(final String source,
             final Type<? extends GridMeasurementPublishedEvent> destinationType, final MappingContext mappingContext) {
 
-        StringArrayToAnalogList stringArrayToAnalogList = null;
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LsMeasurementMessageToAnalogList stringArrayToAnalogList = null;
+        LOGGER.info("Source string: {}", source);
+
+        try {
+            final Payload payload = this.objectMapper.readValue(source, Payload[].class)[0];
+
+            final String[] values = (payload.gisnr + ", " + String.join(", ", payload.data)).split(", ");
+            LOGGER.info("Values length: {} and values: {}", values.length, Arrays.toString(values));
```

Comment:
I would make this a debug message instead of info. Also without knowing the java code the message is unclear. Also the only thing this messsage makes clear is that there is (or is not) a bug in the code, since  a few lines above, the Source string is already logged. In general, I think you should remove log messages that are only added to the code to make development itself easier

Reply:
Maybe add in the error message below the number of elements of 'value' and the number of expected elements